server/data/tokenStore.test.ts

```diff
@@ -1,34 +1,66 @@
-import { RedisClient } from 'redis'
+import nock from 'nock'
+import redis, { RedisClient } from 'redis-mock'
 import TokenStore from './tokenStore'
+import config from '../config'
 
-const redisClient = {
-  on: jest.fn(),
-  get: jest.fn(),
-  set: jest.fn(),
-}
+let tokenStore: TokenStore
+let redisClient: RedisClient
 
-describe('tokenStore', () => {
-  let tokenStore: TokenStore
+jest.mock('redis', () => redis)
 
-  beforeEach(() => {
-    tokenStore = new TokenStore(redisClient as unknown as RedisClient)
-  })
+jest.mock('../authentication/clientCredentials', () => {
+  return jest.fn().mockImplementation(() => 'Basic clientId:secret')
+})
 
-  afterEach(() => {
-    jest.resetAllMocks()
+beforeEach(() => {
+  redisClient = redis.createClient()
+  tokenStore = new TokenStore()
+})
+
+afterEach(done => {
+  redisClient.flushall(done)
+  jest.clearAllMocks()
+  nock.cleanAll()
+})
+
+describe('Token Store', () => {
+  it('Token for user is returned from redis is exists', async () => {
+    redisClient.set('joebloggs', 'saved token')
```

Comment:
This test is checking a scenario which won't happen? Only the system api client token is cached in Redis here.

Reply:
This scenario will happen, its testing that the system token is returned from cache instead of generated again